src/azure-cli-core/azure/cli/core/application.py

```diff
@@ -110,7 +110,8 @@ def __init__(self, configuration=None):
             'headers': {},  # the x-ms-client-request-id is generated before a command is to execute
             'command': 'unknown',
             'completer_active': ARGCOMPLETE_ENV_NAME in os.environ,
-            'query_active': False
+            'query_active': False,
+            'interactive': False
```

Comment:
'interactive' could just be interpreted as "TTY available", which would be a misinterpretation.  Consider a more verbose but self-documenting name like "az_interactive_active". This makes it less likely a developer would see this in the future and use it for unintended purposes.